app/app/auth.py

```diff
@@ -0,0 +1,41 @@
+from flask import Blueprint, render_template, request, redirect, url_for, flash
+from .models import User
+from werkzeug.security import generate_password_hash, check_password_hash
+from . import db
+from flask_login import login_user, login_required, logout_user, current_user
+
+MIN_USERNAME_LENGTH = 4
+MIN_PASSWORD_LENGTH = 7
+
+auth = Blueprint('auth', __name__)
+
+@auth.route('/login', methods=['GET', 'POST'])
+def admin_login():
+    if current_user.is_authenticated:
+        user = User.query.filter_by(id = current_user.id).first()
+        theme = user.selected_theme()
+    else:
+        theme = "Blue"
+
+    if request.method == 'POST':
+        username = request.form.get('username')
+        password = request.form.get('password')
+
+        user = User.query.filter_by(username=username).first()
+        if user:
+            if user.check_password(password):
+                flash("Logged in successfully! Hello " + user.username, category="success")
+                login_user(user, remember=True)
+                return redirect(url_for('routes.learn'))
+            else:
+                flash("Incorrect password, try again.", category="error")
+        else:
+            flash("User does not exist", category="error")
+
+    return render_template('login.html', user=current_user, theme=theme)
+
+@auth.route('/logout')
+@login_required
+def logout():
+    logout_user()
```

Comment:
I won't be reviewing this part as I will assume you will continue the work from #30